homeassistant/components/velbus/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for the Velbus platform."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PORT, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.util import slugify
+
+from .const import DOMAIN
+
+
+@callback
+def velbus_entries(hass: HomeAssistant):
+    """Return connections for Velbus domain."""
+    return set((entry.data[CONF_PORT]) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class VelbusConfigFlow(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self) -> None:
+        self._errors = {}
+
+    async def _create_device(self, name: str, prt: str):
+        return self.async_create_entry(
+            title=name,
+            data={
+                CONF_PORT: prt
+            }
+        )
+
+    def _prt_in_configuration_exists(self, prt: str) -> bool:
+        """Return True if port exists in configuration."""
+        if prt in velbus_entries(self.hass):
+            return True
+        return False
+
+    async def async_step_user(self, user_input=None):
+        """Step when user intializes a integration"""
+        self._errors = {}
+        if user_input is not None:
+            name = slugify(user_input[CONF_NAME])
+            prt = user_input.get(CONF_PORT)
+            # name must be unique
+            if not self._prt_in_configuration_exists(prt):
+                return await self._create_device(name, prt)
+            self._errors[CONF_NAME] = 'port_exists'
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required(CONF_NAME): str,
+                vol.Required(CONF_PORT): str
+            }),
+            errors=self._errors
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Import a config entry."""
+        prt = user_input.get(CONF_PORT)
+        name = user_input.get(CONF_NAME)
+        if self._prt_in_configuration_exists(prt):
+            # if the velbus import is already in the config
+            # we should not proceed the import
+            return None
+        return await self._create_device(name, prt)
```

Comment:
Probably pass the data to the user step here instead.

Reply:
We either abort or pass to the user_step, in this case i would rather abort as this means we already imported from the config file.
I don't want to end up re-importing the config file every time we restart

Reply:
I meant that we should pass to the user step instead of creating the entry directly.

Reply:
this means that the import step will do nothing more then just passing all info into the user step?

as all tests that are done in the user step are  already performed we end up just creating the device, why pass it to the user step then?

Reply:
Yes, we want to have the same main flow for both user and import steps entry points. The config flow can be started either from the user step or from the import step. In both cases we want to check the we can connect to the device before creating the config entry. Having only one of the steps with this logic saves code.